File: test-resources-kafka/src/test/java/io/micronaut/testresources/kafka/SchemaRegistryClientConfiguration.java
package io.micronaut.testresources.kafka;

// tag::clazz[]
import io.micronaut.context.annotation.Requires;
import io.micronaut.context.annotation.Value;
import jakarta.inject.Singleton;

@Singleton
@Requires(property = "kafka.schema.registry.url")
class SchemaRegistryClientConfiguration {

    SchemaRegistryClientConfiguration(
            @Value("${kafka.schema.registry.url}") String schemaRegistryUrl) {
        // Configure a Schema Registry client for tests.
    }
}
// end::clazz[]

File: test-suite-docs-python/src/test/python/micronaut/testresources/docs/kafka/SchemaRegistryClientConfiguration.py
# tag::imports[]
from typing import Annotated

from jakarta.inject import Singleton
from micronaut.context.annotation import Requires, Value
# end::imports[]


# tag::clazz[]
@Singleton
@Requires(property="kafka.schema.registry.url")  # <1>
class SchemaRegistryClientConfiguration:

    def __init__(self, schema_registry_url: Annotated[str, Value("${kafka.schema.registry.url}")]):  # <2>
        # Configure a Schema Registry client for tests.
        self.schema_registry_url = schema_registry_url
# end::clazz[]
